frontend/lib/src/hooks/useDebouncedCallback.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useCallback, useEffect, useRef } from "react"
+
+/**
+ * Interface for the return value of the useDebouncedCallback hook.
+ */
+interface UseDebouncedCallbackReturn<A extends any[]> {
```

Comment:
**suggestion:** For better type safety can we utilize `A extends unknown[]`?

Reply:
Sounds good 👍  Updated.